boot/boot_test.go

```diff
@@ -77,21 +77,183 @@ func (s *bootenvSuite) SetUpTest(c *C) {
 	s.forceBootloader(s.bootloader)
 }
 
-type bootenv20Suite struct {
+type baseBootenv20Suite struct {
 	baseBootenvSuite
 
+	kern1 snap.PlaceInfo
+	kern2 snap.PlaceInfo
+	base1 snap.PlaceInfo
+	base2 snap.PlaceInfo
+
+	normalDefaultState      *uc20bootStateSetupOpts
+	normalTryingKernelState *uc20bootStateSetupOpts
+}
+
+func (s *baseBootenv20Suite) SetUpTest(c *C) {
+	s.baseBootenvSuite.SetUpTest(c)
+
+	var err error
+	s.kern1, err = snap.ParsePlaceInfoFromSnapFileName("pc-kernel_1.snap")
+	c.Assert(err, IsNil)
+	s.kern2, err = snap.ParsePlaceInfoFromSnapFileName("pc-kernel_2.snap")
+	c.Assert(err, IsNil)
+
+	s.base1, err = snap.ParsePlaceInfoFromSnapFileName("core20_1.snap")
+	c.Assert(err, IsNil)
+	s.base2, err = snap.ParsePlaceInfoFromSnapFileName("core20_2.snap")
+	c.Assert(err, IsNil)
+
+	// default boot state for robustness tests, etc.
+	s.normalDefaultState = &uc20bootStateSetupOpts{
+		modeenv: &boot.Modeenv{
+			// base is base1
+			Base: s.base1.Filename(),
+			// no try base
+			TryBase: "",
+			// base status is default
+			BaseStatus: boot.DefaultStatus,
+			// current kernels is just kern1
+			CurrentKernels: []string{s.kern1.Filename()},
+			// operating mode is run
+			Mode: "run",
+			// RecoverySystem is unset, as it should be during run mode
+			RecoverySystem: "",
+		},
+		// enabled kernel is kern1
+		kern: s.kern1,
+		// no try kernel enabled
+		tryKern: nil,
+		// kernel status is default
+		kernStatus: boot.DefaultStatus,
+	}
+
+	// state for after trying a new kernel for robustness tests, etc.
+	s.normalTryingKernelState = &uc20bootStateSetupOpts{
+		modeenv: &boot.Modeenv{
+			// base is base1
+			Base: s.base1.Filename(),
+			// no try base
+			TryBase: "",
+			// base status is default
+			BaseStatus: boot.DefaultStatus,
+			// current kernels is kern1 + kern2
+			CurrentKernels: []string{s.kern1.Filename(), s.kern2.Filename()},
+		},
+		// enabled kernel is kern1
+		kern: s.kern1,
+		// try kernel is kern2
+		tryKern: s.kern2,
+		// kernel status is trying
+		kernStatus: boot.TryingStatus,
+	}
+}
+
+type bootenv20Suite struct {
+	baseBootenv20Suite
+
 	bootloader *bootloadertest.MockExtractedRunKernelImageBootloader
 }
 
+
+var defaultUC20BootEnv = map[string]string{"kernel_status": boot.DefaultStatus}
+
 var _ = Suite(&bootenv20Suite{})
 
 func (s *bootenv20Suite) SetUpTest(c *C) {
-	s.baseBootenvSuite.SetUpTest(c)
+	s.baseBootenv20Suite.SetUpTest(c)
 
 	s.bootloader = bootloadertest.Mock("mock", c.MkDir()).WithExtractedRunKernelImage()
 	s.forceBootloader(s.bootloader)
 }
 
+
+type uc20bootStateSetupOpts struct {
+	modeenv    *boot.Modeenv
+	kern       snap.PlaceInfo
+	tryKern    snap.PlaceInfo
+	kernStatus string
+}
+
+func setupUC20Bloader(
```

Comment:
given this writes to modeenv Bloader seems a bit too specific.   setupUC20Bootenv  (that's what most suite are called after after all)